src/Transform/ONNX/ReplaceNoValueOperands.cpp

```diff
@@ -86,8 +86,10 @@ class GemmReplaceNoValue : public OpRewritePattern<ONNXGemmOp> {
     if (isNotNoValue(matrixC)) {
       return rewriter.notifyMatchFailure(op, "has no NoValue operand");
     }
+    mlir::Value B = op.B();
+    ArrayRef<int64_t> cshape(B.getType().cast<TensorType>().getShape()[0]);
```

Comment:
We probably want to assert or check that `B` has an appropriate number of elements. Not sure about `NoneType`. If this is not helpful please ignore. Did not check the TOSA conversions for a while.

Reply:
It was really helpful because it made me find a bug where if the `transB` is set I had to take the second dimension etc.. (So I swapped it to use the results' second dimension instead.